rust/ql/lib/ide-contextual-queries/Definitions.qll

```diff
@@ -0,0 +1,117 @@
+/**
+ * Provides classes and predicates related to jump-to-definition links
+ * in the code viewer.
+ */
+
+private import codeql.rust.elements.Variable
+private import codeql.rust.elements.Locatable
+private import codeql.rust.elements.FormatArgsExpr
+private import codeql.rust.elements.FormatArgsArg
+private import codeql.rust.elements.Format
+private import codeql.rust.elements.NamedFormatArgument
+private import codeql.rust.elements.PositionalFormatArgument
+private import codeql.Locations
+
+/** An element with an associated definition. */
+abstract class Use extends Locatable {
```

Comment:
It probably makes sense to restrict this to uses that are not the result of a macro expansion?

Reply:
Not sure I understand. This PR adds jump to definition for format arguments, which are all uses that are the result of macro expansion. 

Perhaps we should restrict the definitions though. That should get rid of all the `left_val` and `right_val` references in the expansion of `assert!` macros.